### PR TITLE
Define extra options for properties 

### LINKED ISSUE #666: [REQUEST] Improve strict schema validation
## Is your feature request related to a problem? Please describe.
https://github.com/rswag/rswag/pull/604#issuecomment-1641941197

## Describe the solution you'd like
Use `allPropertiesRequired` and `noAdditionalProperties` options for json-schema gem implemented in https://github.com/voxpupuli/json-schema/pull/494.

## Relates to which version of OAS (OpenAPI Specification)
- OAS3
- OAS3.1

### PR DESCRIPTION
**Pull request by IlkhamGaysin**
## Problem
It closes https://github.com/rswag/rswag/issues/666

## Solution
Added two more options to allow to use `allPropertiesRequired` and `noAdditionalProperties` options from [json-schema](https://github.com/a-lavis/json-schema)

### This concerns this parts of the OpenAPI Specification:
* [EXAMPLE_LINK TO RELEVANT OPEN API SPECS PAGE](https://spec.openapis.org/oas/v3.1.0#data-types)
* [ANOTHER LINK TO RELEVANT OPEN API SPECS PAGE](https://spec.openapis.org/oas/v3.1.0#schema)

### The changes I made are compatible with:
- [ ] OAS2
- [ ] OAS3
- [ ] OAS3.1

### Related Issues
Links to any related issues.

### Checklist
- [x] Added tests
- [ ] Changelog updated
- [ ] Added documentation to README.md
- [ ] Added example of using the enhancement into [test-app](https://github.com/rswag/rswag/tree/master/test-app)

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by mustmodify** [comment]
@IlkhamGaysin what's the status of this ticket?

**Comment by IlkhamGaysin** [comment]
@mustmodify updated with master let's see what CI will give us

**Comment by IlkhamGaysin** [comment]
@romanblanco could you please approve workflow to run on the CI 🍺

**Comment by Kulgar** [comment]
Would love to see this one added to the project - can I help?

**Comment by IlkhamGaysin** [comment]
@Kulgar Could someone merge this? I don't see CI results on this PR.  :/

**Comment by romanblanco** [comment]
> @romanblanco could you please approve workflow to run on the CI 🍺

Will try to look into why the CI is not running on the PR. Do the tests pass localy?

File: test-app/spec/integration/blogs_spec.rb [hunk 1]
```diff
@@ -165,6 +173,30 @@
...
+        run_test!
+      end
+
+      response '200', 'blog found - openapi_all_properties_required = true', openapi_all_properties_required: true do
+        schema '$ref' => '#/components/schemas/blog'
+
+        let(:id) { blog.id }
+
+        run_test!
+      end
```
Comment (romanblanco) [inline test-app/spec/integration/blogs_spec.rb:191]: These specs are identical

File: rswag-specs/spec/rswag/specs/configuration_spec.rb [hunk 2]
```diff
@@ -189,4 +196,32 @@
...
+    let(:response) { subject.openapi_all_properties_required }
+
+    context 'when not provided' do
+      let(:openapi_all_properties_required) { nil }
+      it { expect(response).to eq(false) }
+    end
+
+    context 'when provided in rspec config' do
+      let(:openapi_all_properties_required) { true }
+       it { expect(response).to eq(true) }
```
Comment (romanblanco) [inline rswag-specs/spec/rswag/specs/configuration_spec.rb:210]: Nitpick: indentation

**Review by romanblanco — CHANGES_REQUESTED** [review]

**Comment by a-lavis** [comment]
@romanblanco I opened #783 to address the above feedback since I don't have write access to this branch.  I hope its helpful - feel free to close if not.

**Pull request merged**